File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetPhysicalDeviceQueueFamilyProperties.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetPhysicalDeviceQueueFamilyProperties_callNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jlong pQueueFamilyPropertyCount, jlong pQueueFamilyProperties){
    (void)env;
    (void)jc;
    PFN_vkGetPhysicalDeviceQueueFamilyProperties f = (PFN_vkGetPhysicalDeviceQueueFamilyProperties)jniLongToFunctionPointer(address);
    f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        ((uint32_t*)jniLongToPointer(pQueueFamilyPropertyCount)),
        ((VkQueueFamilyProperties*)jniLongToPointer(pQueueFamilyProperties))
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkGetPhysicalDeviceQueueFamilyProperties.java
package cz.mg.vulkan;

public class PFNvkGetPhysicalDeviceQueueFamilyProperties extends VkFunctionPointer {
    public PFNvkGetPhysicalDeviceQueueFamilyProperties() {
    }

    protected PFNvkGetPhysicalDeviceQueueFamilyProperties(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetPhysicalDeviceQueueFamilyProperties(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetPhysicalDeviceQueueFamilyProperties(long value) {
        setValue(value);
    }

    public PFNvkGetPhysicalDeviceQueueFamilyProperties(VkInstance instance) {
        super(instance, new VkString("vkGetPhysicalDeviceQueueFamilyProperties"));
    }

    public void call(VkPhysicalDevice physicalDevice, VkUInt32 pQueueFamilyPropertyCount, VkQueueFamilyProperties pQueueFamilyProperties){
        callNative(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), pQueueFamilyPropertyCount != null ? pQueueFamilyPropertyCount.getVkAddress() : VkPointer.NULL, pQueueFamilyProperties != null ? pQueueFamilyProperties.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long physicalDevice, long pQueueFamilyPropertyCount, long pQueueFamilyProperties);

}
